### ISSUE TITLE
Provide RPM and DEB packaging

### ISSUE THREAD

**Issue by danoliv** [lesfurets/git-octopus#24]
I work for Kelkoo. We would like to use this project, but we need to repackage with rpm and maybe submit to fedora. 
Did you already do it or plan to do it? Otherwise we can.
Thanks for the great job.

**Comment by apflieger** [comment]
Absolutely great. We never did it and always wanted to.
If you need any precisions on how to build the sources or need some changes on the project, let us know :)

**Comment by mathieubolla** [comment]
Hi,

I heard it may be possible to have RPMs built on top of the Homebrew port. I don't remember how, but have a look at Linux Homebrew project, they may already have what you want...

Mathieu (git-octopus Homebrew (macOs) maintainer)

**Comment by troisdiz** [comment]
I think that if you do it using fpm you will get both rpm and deb packaging

**Comment by lesfurets-admin** [comment]
I can contact Henri Gomez or Pierre Antoine Gregoire on this if needed.
I’ll be piece of cake for them ☺

Dimitri BAELI
IT Associate Director

[cid:image001.png@01D24191.EA18C180]https://www.lesfurets.com/
12-14 rue Médéric
75017 Paris
+33 1 56 02 36 50
LesFurets.comhttps://www.lesfurets.com/
Facebookhttps://www.facebook.com/hervelesfurets - Twitterhttps://twitter.com/lesfurets

**Comment by danoliv** [comment]
Thanks for the support. If nobody already did we can work on this we have experience on rpm packaging and we have allocated some time for this. We don't use .deb but could have a look at fpm.

Are you ok if we submit to fedora?
Do you prefer to be the maintainer of the package?

**Comment by apflieger** [comment]
Nope, that's fine.

**Comment by dubreuia** [comment]
We should provide a .deb package too... I'll look into it when I have time.

**Comment by danoliv** [comment]
Hi everybody we submitted a review request for fedora : https://bugzilla.redhat.com/show_bug.cgi?id=1401013#c1, cross the fingers!
If someone wants to install the rpm (el7,f24,f25,f26) you can get it here : https://copr.fedorainfracloud.org/coprs/baitaand/git-octopus/build/484097/ 
The spec file is temporary here : https://github.com/danoliv/git-octopus-spec. If the package will be approved it will be hosted on a fedora git repository.

**Comment by apflieger** [comment]
woohoo !

**Comment by dubreuia** [comment]
Can we add that to the documentation? Should we wait it is hosted the fedora git repo? And how does it work when we release a new version?

**Comment by danoliv** [comment]
Yes you can add it in the documentation, I will improve the doc on github so you can use it as reference. 

Usually it is the maintainer work to prepare the rpm when a new version of the software is available, a ticket will be opened automatically in the fedora bugtracker as soon as a new tar.gz is available on github.

All this will be available after the package will be approved. 

In the mean time the copr repository can be used to install the rpm, you can create an "official one" if you want to or use mine https://copr.fedorainfracloud.org/coprs/baitaand/git-octopus/.

**Comment by dubreuia** [comment]
OK I'll link your copr repository in the README and close this issue when the package is approved. Thanks!

**Comment by danoliv** [comment]
In this case you can add this to the README:
dnf copr enable baitaand/git-octopus 
dnf install git-octopus

**Comment by dubreuia** [comment]
What do we do when there's a new git-octopus version? Can you give me steps to release the rpm?

That makes me think we'll need our own copr repository if we don't want to be dependant on you

**Comment by dubreuia** [comment]
Added a section in the README https://github.com/lesfurets/git-octopus#rpm-package-manager

I'm working on the .deb script I'll commit it in the repo (https://www.debian.org/doc/manuals/debmake-doc/ch04.en.html)

**Comment by apflieger** [comment]
There's a [RELEASE.md](https://github.com/lesfurets/git-octopus/blob/master/RELEASE.md) file. It would be nice to mention fedora's release process in it.

**Comment by danoliv** [comment]
In order to submit to fedora you need to became an official fedora maintainers (https://fedoraproject.org/wiki/Join_the_package_collection_maintainers). It is a long process, the person needs to pass a sort of screening, review or produce other packages and find a sponsor that support its application.

After the person is approved it can be the official maintainer (or the co-maintainer) of the package. At Kelkoo we are integrating git-octopus in our workflow so we are motivated to maintain/co-maintain the rpm.

I started the process by submitting the spec file for git-octopus for a review https://bugzilla.redhat.com/show_bug.cgi?id=1401013, see https://fedoraproject.org/wiki/Package_Review_Process to understand the process. 

Usually the good practice is to have a separated repository for the packaging (spec files, patch, etc) and fedora will provide a git repository as soon as the package and the maintainer are approved.

I will update the documentation as soon as possible.

**Comment by apflieger** [comment]
Again, thank a lot for doing that.
By the way, did you looked at https://github.com/lesfurets/git-octopus/issues/25?
I'm rewriting git-octopus in go, so the packaging will change. I think it will just be simplier

**Comment by danoliv** [comment]
Yes, pretty cool! As soon as you have a beta version I would like to test it and to look at the packaging.

**Comment by danoliv** [comment]
I want to announce that git-octopus (version 1.4) is on the fedora official repository : https://admin.fedoraproject.org/pkgdb/package/rpms/git-octopus/

The package is in testing at the moment, you can install it by doing: 
dnf --enablerepo=updates-testing install git-octopus

**Comment by apflieger** [comment]
Woohoo \o/

**Comment by dbaeli** [comment]
Thank you !

Dimitri BAELI

Le 21 mars 2017 à 15:51, Arnaud Pflieger <notifications@github.com> a écrit :

Woohoo \o/

**Comment by danoliv** [comment]
I have a question about the packaging: at the moment the minimum git version is 1.8, is there any particular reason for excluding git 1.7?
Thanks!

**Comment by apflieger** [comment]
[This change](https://github.com/lesfurets/git-octopus/pull/17/files#diff-21be0d50914712e7cf65478dbc2ea1baR126) introduced in 1.4 that fixes #16 uses an option of git merge-base that is available only on git >= 1.8
For git-octopus 2.0 though, this is not required. The case is handled without this command. The unit test `TestOctopusAlreadyUpToDate` checks this behavior.

**Comment by danoliv** [comment]
Great! so from git-octopus 2 it can be compatible with git 1.7 (therefore with centos 6). 
Thanks!

**Comment by apflieger** [comment]
Probably. Which version of git-octopus do you use right now ?

**Comment by danoliv** [comment]
2.0 beta1 from our private rpm repository, soon the beta2
We are waiting for the final 2.0 release to put in the officials centos and fedora repos.

**Comment by apflieger** [comment]
So you are using 2.0 already, that's good news!
The problem for the release is that we still miss a bunch of features that was implemented in 1.x
